images/router/haproxy/conf/haproxy-config.template

```diff
@@ -33,6 +33,11 @@
 
 global
   maxconn {{env "ROUTER_MAX_CONNECTIONS" "20000"}}
+{{- if gt (env "ROUTER_THREADS") "0" }}
```

Comment:
Because it is operating on strings (not numbers), `gt` performs a lexical byte-wise comparison in this expression, which (a) is strange enough to suggest it is not the intention and (b) means that the expression will evaluate true for such values of `ROUTER_THREADS` as "00", "off", or weird Unicode variants of zero.  It would be more correct to do something like `if ne "" (firstMatch "[1-9][0-9]*" (env "ROUTER_THREADS"))`.  Alternatively, it would be no less correct (compared to using `gt`) and possibly less misleading to do something like `if ne "0" (env "ROUTER_THREADS" "0")`.

Reply:
What about adding more fields to the data structure on which the template is applied? I'm not sure what advantage environment offers here, but can think of many disadvantages. For example, we could add:

```go
type templateData struct {
  // ...
  RouterThreads uint
}
```

and then the `gt` should work.

We could also extract the decision into the Go code for even more clarity:

```go
type templateData struct {
  // ...
  RouterThreads uint
  RouterThreadsEnabled bool
}
```
```
{{ if .RouterThreadsEnabled }}
nbthread {{ .RouterThreads }}
{{ end }}
```

Reply:
The preference in the past from @smarterclayton has been to prefer using the ENV.  Adding fields to the object, or new functions, are more strictly considered part of the API.

Reply:
I'm surprised by that given the type being used in the template isn't actually a versioned API object... and the env variables themselves represent an implicit API (one which is much harder to deal with).

Reply:
If we're stuck with environment for now, then I have no objection to @Miciah's black template magic